apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Tests/Tasks/TestScheduler.cs

```diff
@@ -148,169 +148,213 @@ void ActionWrapper()
 
         private class TestTaskScheduler : TaskScheduler, IDisposable
         {
+            [ThreadStatic]
```

Comment:
I'm skipping reviewing this for now, in order to look at the prod code. Happy to come back to this later, but may need a chat first.